docker/Dockerfile

```diff
@@ -24,15 +24,25 @@
 #     -t the target repo and tag name
 #     more options can be found with -h
 
-ARG BASE_IMAGE=8-jre-slim
-FROM openjdk:${BASE_IMAGE}
+ARG BASE_IMAGE=openjdk:8-jre-slim
+ARG spark_provided="spark_builtin"
 
-ARG kyuubi_uid=10009
+FROM ${BASE_IMAGE} as builder_spark_provided
+
+FROM ${BASE_IMAGE} as builder_spark_builtin
+
+ONBUILD ENV SPARK_BINARY /opt/spark
+ONBUILD RUN mkdir -p  ${SPARK_BINARY}
+ONBUILD COPY spark-binary ${SPARK_BINARY}
```

Comment:
do you need `SPARK_BINARY` ? seems we can just `COPY spark-binary ${spark_home}`

Reply:
OK, good idea. I will advance the `ARG spark_home`.